hal_st/stm32fxxx/DmaStm.cpp

```diff
@@ -529,10 +538,13 @@ namespace hal
     void DmaStm::Stream::Disable()
     {
         auto streamRegister = DmaChannel[dmaIndex][streamIndex];
-#if defined(DMA_SxCR_EN)
-        streamRegister->CR &= ~DMA_SxCR_EN;
-#else
+#ifdef GPDMA1
+        streamRegister->CCR |= (DMA_CCR_SUSP | DMA_CCR_RESET);
         streamRegister->CCR &= ~DMA_CCR_EN;
+#elif defined(DMA_CCR_EN)
+        streamRegister->CCR &= ~DMA_CCR_EN;
+#else
+        streamRegister->CR &= ~DMA_SxCR_EN;
 #endif
```

Comment:
I think this would be equivalent, but saves a duplication?

```suggestion
#ifdef GPDMA1
        streamRegister->CCR |= (DMA_CCR_SUSP | DMA_CCR_RESET);
#endif

#if defined(DMA_CCR_EN)
        streamRegister->CCR &= ~DMA_CCR_EN;
#else
        streamRegister->CR &= ~DMA_SxCR_EN;
#endif
```

Reply:
Nice improve, thanks.